expression/builtin_time.go

```diff
@@ -51,7 +51,11 @@ const ( // GET_FORMAT location.
 )
 
 // DurationPattern determine whether to match the format of duration.
-var DurationPattern = regexp.MustCompile(`^(|[-]?)(|\d{1,2}\s)(\d{2,3}:\d{2}:\d{2}|\d{1,2}:\d{2}|\d{1,6})(|\.\d*)$`)
+//var DurationPattern = regexp.MustCompile(`^(|[-]?)(|\d{1,2}\s)(\d{2,3}:\d{2}:\d{2}|\d{1,2}:\d{2}|\d{1,6})(|\.\d*)$`)
+var DurationPattern = regexp.MustCompile(`^\s*[-]?(((\d{1,2}\s+)?0*\d{0,3}(:0*\d{1,2}){0,2})|(\d{1,7}))?(\.\d*)?\s*$`)
+
+// TimestampPattern determine whether to match the format of timestamp.
```

Comment:
Have to use regular expression? It is unreadable and has poor efficiency.

Reply:
The legal format for the literals are quite complicated. For example, for `DATE` literal, some weird format  such as `00002017-0000012-00031`, `1@002@0009` are all legal. That;s why I use  regular expression here to check these. 
I think the efficiency is not a big problem because usually the sql using these literals don't have a big result set ( usually just a one-row return sentence).  I think there's no need to make it complicated for the rare condition where the efficiency do matters.